### ISSUE TITLE
semantic: `is_reference_to_global_variable` doesn't return expected value in specific case.

### ISSUE THREAD

**Issue by tbashiyy** [oxc-project/oxc#8217]
## What is this
[`Semantic.is_reference_to_global_variable`](https://github.com/tbashiyy/oxc/blob/cd349a312d509b89534a23ed64aea849750eb447/crates/oxc_semantic/src/lib.rs#L216-L218) doesn't return expected value in case of below.

**js source code**
```javascript
Object = {}
console.log(Object);
```

When overriding global variable without declarations, I think this is no longer global variables.
But when checking `Object identifier` of  `console.log` argument by using `is_reference_to_global_variable`, it returns `true`. 

I think this is because of `Object` is still in `root_unresolved_references`.

## What I want
In this case, `is_reference_to_global_variable` returns `false`.
Or I want to know another way to check whether identifier of global variable is overridden or not.

**Comment by Boshen** [comment]
Your example is still a [global variable](https://developer.mozilla.org/en-US/docs/Glossary/Global_variable), it is not a global variable when it's shadowed by `let`. Assigning to it is still a global variable.

It seems like we don't have a easy way to detect this right now, although the rule `no-global-assign` will report a violation in this case.
